saas_demo/os.py

```diff
@@ -120,6 +125,84 @@ def update_config(section, key, value):
         config_parser.write(configfile)
 
 
+def _fileopen(path, mode, basedir, pathinfo):
+    name = os.path.normpath(os.path.normcase(os.path.join(basedir, path)))
+    if os.path.isfile(name):
+        if 'b' in mode:
+            fo = open(name, mode)
+        else:
+            fo = io.open(name, mode, encoding='utf-8')
+        if pathinfo:
+            return fo, name
+        return fo
```

Comment:
too complicated. 

* You should use ``with open (...) as f`` construction. You can make a method to compute filename if needed

Reply:
We can leave it as is, if it's copy-pasted code